File: Unidad_1/mcd/mcd.pl
% Ejemplo de Maximo Comun Divisor en Prolog
% En el paradigma lógico lo que se indican son reglas o predicados
% para posteriormente llevar a cabo una consulta
% En este caso: mcd(69,120,X).

%main:- mcd(69,120,X),writeln(X).

mcd(A,B,G) :- A = B, G = A.
mcd(A,B,G) :- A > B, C is A-B, mcd(C,B,G).
mcd(A,B,G) :- B > A, C is B-A, mcd(C,A,G).

File: Unidad_1/mcd/mcd.c
/**
Ejemplo de Maximo Comun Divisor en C
En paradigma imperativo es usual que el computo se realice como una secuencia
de instrucciones con ciclos y condicionales
*/

#include <stdio.h>

int mcd(int a, int b) {
    while (a != b) {
      if (a > b) a = a - b;
      else b = b - a;
      }
    return a;
}

int main(){
  printf("%d\n",mcd(69,120));
  return 0;
}

